src/lib/pickles/composition_types/branch_data_intf.ml

```diff
@@ -11,6 +11,8 @@ module type S = sig
     end]
 
     val of_int_exn : int -> t
+
+    val to_int : t -> int
```

Comment:
Minor: what's the rationale with the introduction of `to_int`? I have not seen uses of it  (but I might have missed them) so it feels gratuitous.

- if it's for conversion reasons, the we might want more specific functions than `to_int` (if possible)
- if it's for pretty printing reasons, then let's introduce a pretty printer here (e.g., `Format.pp_print_char`) or use the sexp/json functions.

Then, again, my comment might be a little off hence my first question